### DISCUSSION TITLE
Forwarded emails not being ingested?

### DISCUSSION THREAD

**Discussion by HeathL17** [moralmunky/Home-Assistant-Mail-And-Packages#617] [Q&A]
Greetings.

I set up a new GMail mailbox just for package tracking.  I have rules configured on my primary mailbox to forward mails to this mailbox.

I'm not seeing packages tracking data show up despite mails being present in the mailbox.  I enabled debug logging, and it appears to search for mails from specified senders.  In my case, since these emails are forwarded automatically via a rule, I am not sure they will ever find the emails in question.

Given this mailbox is only used for mail and packages, is there a way to bypass the sender searches and only look at subject line?  

Appreciate any help.

H

**Comment by HeathL17** [comment]
Quick update, I have made it work by adding my email address in const.py, but I worry about future updates, etc.

Interested to hear if there is a better path or other workarounds.

**Comment by firstof9** [comment]
> Given this mailbox is only used for mail and packages, is there a way to bypass the sender searches and only look at subject line?

No

Emails forwarded from Gmail -> Gmail should look like they are coming from the original sender.

**Reply by HeathL17** [reply]
Hmmm.  My other account is non-Gmail so that is probably the difference.

Looking through the email from: and subject lines, the sender seems less important than the subject line in the code. Perhaps it is a worthy feature request to enable an input asking for the FWDing email address, similar to the Amazon email address handling?

**Reply by moralmunky** [reply]
Sender email address matches help identify the particular shipping vendor if there ever was a case where two shipping vendors used the same subject verbiage.

**Reply by HeathL17** [reply]
First, thanks for building this.  It's a great integration, and it brings a ton of value to HA.  

This behavior must be GMail specific, then?  I am not seeing the same behavior from emails on between mailboxes on the same platform in the supported list (Hotmail).  I am seeing in the headers that the from address is my other mailbox that I am forwarding from.

It seems odd to me that it works for so many, yet I am running into such a rudimentary issue because I am not using GMail for both sides of the mail flow?  Weird.  

I would imagine most non-GMail folks use only one mailbox, which probably wouldn't manifest these problems.  That's not a great security practice, IMO, given the number of items tied to email for account recoveries, etc. that are now inheriting trust from HA + this integration.  

It would be ideal if there was a global email address that could be appended to all the sender definitions - perhaps even an optional override?  I've hardcoded it into const.py and it seems to work fine on my end.  I'd do the work and submit a PR, but I am not proficient enough in Python to do that, unfortunately.

**Reply by firstof9** [reply]
It would be far easier to just have the shippers send email directly to your "shipping email".

> My other account is non-Gmail so that is probably the difference.

Yes this is a gmail specific feature, I'm not sure why they do this but I like it.

Alternatively, you could just add multiple email accounts by adding multiple integration configurations.

**Reply by HeathL17** [reply]
Appreciate the responses.

> It would be far easier to just have the shippers send email directly to your "shipping email".

I tried that earlier.  Unfortunately, it seems two of my shippers won't let me have shipment info sent tied to more than one email account.

> Alternatively, you could just add multiple email accounts by adding multiple integration configurations.

I don't want HA having access to my mailbox, hence the desire for a separate "shipping email" mailbox.  

I could probably switch it and have the "shipping email" forward to my primary mail account, though.  Not ideal but given the circumstances it's not awful and means I can gracefully take future updates to this integration.

If nothing else, maybe it's worth noting in the Wiki to help others.
